File: backend/app/utils/currency.py
def format_rupiah(amount: int | float | None) -> str:
    """Format an integer or float amount into Indonesian Rupiah (e.g. Rp 8.750)."""
    if amount is None:
        return "Rp 0"
    amount_int = int(round(amount))
    formatted = f"{amount_int:,}".replace(",", ".")
    return f"Rp {formatted}"

File: backend/src/utils/currency.ts
export function formatRupiah(amount: number | null | undefined): string {
  if (amount === null || amount === undefined || isNaN(amount)) {
    return 'Rp 0';
  }
  const amountInt = Math.round(amount);
  const formatted = amountInt.toString().replace(/\B(?=(\d{3})+(?!\d))/g, '.');
  return `Rp ${formatted}`;
}
